src/Serializer/AbstractItemNormalizer.php

```diff
@@ -186,10 +186,19 @@ public function denormalize($data, $class, $format = null, array $context = [])
             if (!$this->serializer instanceof DenormalizerInterface) {
                 throw new LogicException('Cannot denormalize the input because the injected serializer is not a denormalizer');
             }
+            if ($dataTransformer instanceof PreHydrateInputInterface) {
+                $context[static::OBJECT_TO_POPULATE] = $dataTransformer->createInput($inputClass, $context);
+            }
             $denormalizedInput = $this->serializer->denormalize($data, $inputClass, $format, $context);
             if (!\is_object($denormalizedInput)) {
                 throw new \UnexpectedValueException('Expected denormalized input to be an object.');
             }
+            if ($dataTransformer instanceof PreHydrateInputInterface) {
+                unset($context[static::OBJECT_TO_POPULATE]);
+            }
+            if (null !== $objectToPopulate) {
```

Comment:
this condition looks dangerous and should be merged with the previous `$dataTransformer instanceof PreHydrateInputInterface` no?

Reply:
Done